### PR TITLE
BZ1934587: Accessing registry with existing registry project

### PR DESCRIPTION
**Pull request by snarayan-redhat**
This fix is related to bug BZ[1934587](https://bugzilla.redhat.com/show_bug.cgi?id=1934587)

To preview fix for Accessing the registry, click [here](https://deploy-preview-34316--osdocs.netlify.app/openshift-enterprise/latest/registry/accessing-the-registry.html#prerequisites)

To preview fix for Exposing the registry, click [here](https://deploy-preview-34316--osdocs.netlify.app/openshift-enterprise/latest/registry/securing-exposing-registry.html)

@dmage and @ricardomaraschini  - Kindly review the changes.

@wzheng1 - Kindly review the fix

This fix needs to be merged with 4.6, 4.7, 4.8

### CODE REVIEW

File: modules/registry-exposing-secure-registry-manually.adoc [hunk 1]
```diff
@@ -9,7 +9,7 @@
 Instead of logging in to the {product-title} registry from within the cluster,
 you can gain external access to it by exposing it with a route. This allows you
 to log in to the registry from outside the cluster using the route address, and
-to tag and push images using the route host.
+to tag and push images using the route host of your project.
```
Comment (ricardomaraschini) [inline modules/registry-exposing-secure-registry-manually.adoc:12]: "to tag and push images to an existing project using the route host."

File: registry/accessing-the-registry.adoc [hunk 2]
```diff
@@ -32,13 +32,15 @@ the user must have the `registry-viewer` role. To add this role:
...
-the user must have the `registry-editor` role. To add this role:
+* For writing or pushing images, for example when using the `podman push` command:
+** The user must have the `registry-editor` role. To add this role:
 +
 [source,terminal]
 ----
 $ oc policy add-role-to-user registry-editor <user_name>
 ----
++
+** You must have an existing registry project.
```
Comment (ricardomaraschini) [inline registry/accessing-the-registry.adoc:43]: You must have an "existing project", not an "existing registry project". I would rephrase that to make things clear:

** You can only push to a project that already exists in your cluster.

**Review by ricardomaraschini — CHANGES_REQUESTED** [review]

File: modules/registry-exposing-secure-registry-manually.adoc (hunk 1 above)
Comment (snarayan-redhat) [reply modules/registry-exposing-secure-registry-manually.adoc]: Thanks @ricardomaraschini . I have made the suggested changes.

File: registry/accessing-the-registry.adoc (hunk 2 above)
Comment (snarayan-redhat) [reply registry/accessing-the-registry.adoc]: Thanks @ricardomaraschini . I have made the suggested changes.

**Comment by wzheng1** [comment]
/lgtm
Acked by QE.

File: modules/registry-exposing-secure-registry-manually.adoc [hunk 3]
```diff
@@ -9,7 +9,7 @@
 Instead of logging in to the {product-title} registry from within the cluster,
 you can gain external access to it by exposing it with a route. This allows you
 to log in to the registry from outside the cluster using the route address, and
-to tag and push images using the route host.
+to tag and push images to an existing project using the route host..
```
Comment (ricardomaraschini) [inline modules/registry-exposing-secure-registry-manually.adoc:12]: there is an extra "." in there.

**Review by ricardomaraschini — CHANGES_REQUESTED** [review]

File: modules/registry-exposing-secure-registry-manually.adoc (hunk 3 above)
Comment (snarayan-redhat) [reply modules/registry-exposing-secure-registry-manually.adoc]: Removed.

Comment (snarayan-redhat) [reply modules/registry-exposing-secure-registry-manually.adoc]: > there is an extra "." in there.

Removed.

**Review by ricardomaraschini — APPROVED** [review]
/lgtm

File: modules/registry-exposing-secure-registry-manually.adoc [hunk 4]
```diff
@@ -9,7 +9,7 @@
 Instead of logging in to the {product-title} registry from within the cluster,
 you can gain external access to it by exposing it with a route. This allows you
 to log in to the registry from outside the cluster using the route address, and
-to tag and push images using the route host.
+to tag and push images to an existing project using the route host.
```
Comment (maxwelldb) [inline modules/registry-exposing-secure-registry-manually.adoc:12]: "By using" the route host or using route host information as part of the tag + push process?

ISG:

> ⚠️ using _noun_
> To avoid ambiguity, replace this gerund with either "by using" or "that uses".
> 
> To refer to something "using" one thing or another is often ambiguous. For example, the sentence "Back up the server using ABC" might mean "Back up the server by using ABC" or "Back up the server that uses ABC". This advice also applies to gerunds other than "using". For more information, see the style and grammar tips in the Global audiences topic in IBM Style and the section about "Clarity" in "Developing Quality Technical Information".

File: registry/accessing-the-registry.adoc [hunk 5]
```diff
@@ -32,13 +32,15 @@ the user must have the `registry-viewer` role. To add this role:
 $ oc policy add-role-to-user registry-viewer <user_name>
 ----
 
-* For writing or pushing images, for example when using the `podman push` command,
-the user must have the `registry-editor` role. To add this role:
+* For writing or pushing images, for example when using the `podman push` command:
```
Comment (maxwelldb) [inline registry/accessing-the-registry.adoc:35]: You could avoid nesting these with some restructuring. ISG allows substeps down to two levels, though.

File: registry/accessing-the-registry.adoc [hunk 6]
```diff
@@ -32,13 +32,15 @@ the user must have the `registry-viewer` role. To add this role:
...
+** The user must have the `registry-editor` role. To add this role:
```
Comment (maxwelldb) [inline registry/accessing-the-registry.adoc:36]:
```suggestion
** The user must have the `registry-editor` role. To add this role, enter:
```

ISG requires language like this when asking the user to enter a command. ("Commands, parameters, and options in instructions," p. 196, latest version)

File: registry/accessing-the-registry.adoc [hunk 7]
```diff
@@ -32,13 +32,15 @@ the user must have the `registry-viewer` role. To add this role:
...
 +
 [source,terminal]
 ----
 $ oc policy add-role-to-user registry-editor <user_name>
 ----
++
+** Your cluster must have an existing project where the images can be pushed to.
```
Comment (maxwelldb) [inline registry/accessing-the-registry.adoc:43]: This could just be a note instead of its own bullet. 

```suggestion
+
[NOTE]
====
Your cluster must have an existing project where the images can be pushed to.
====
```

**Review by maxwelldb — CHANGES_REQUESTED** [review]
Looks okay. Picked a couple of nits. Only L36 is a must have.

File: registry/accessing-the-registry.adoc (hunk 7 above)
Comment (snarayan-redhat) [reply registry/accessing-the-registry.adoc:43]: As per my discussion with SME, this is an explicit prerequisite for performing a push. Hence, I avoided making this as note.

File: registry/accessing-the-registry.adoc (hunk 6 above)
Comment (snarayan-redhat) [reply registry/accessing-the-registry.adoc]: Used 'run the following command' adhering to ISG

File: modules/registry-exposing-secure-registry-manually.adoc (hunk 4 above)
Comment (snarayan-redhat) [reply modules/registry-exposing-secure-registry-manually.adoc]: I have made this change as this does sound a bit ambiguous.

Comment (snarayan-redhat) [reply modules/registry-exposing-secure-registry-manually.adoc]: > "By using" the route host or using route host information as part of the tag + push process?
> 
> ISG:
> 
> > warning using _noun_
> > To avoid ambiguity, replace this gerund with either "by using" or "that uses".
> > To refer to something "using" one thing or another is often ambiguous. For example, the sentence "Back up the server using ABC" might mean "Back up the server by using ABC" or "Back up the server that uses ABC". This advice also applies to gerunds other than "using". For more information, see the style and grammar tips in the Global audiences topic in IBM Style and the section about "Clarity" in "Developing Quality Technical Information".

I have made this change as this does sound a bit ambiguous.

File: registry/accessing-the-registry.adoc (hunk 6 above)
Comment (snarayan-redhat) [reply registry/accessing-the-registry.adoc]: > ISG requires language like this when asking the user to enter a command. ("Commands, parameters, and options in instructions," p. 196, latest version)

Used 'run the following command' adhering to ISG

Comment (snarayan-redhat) [reply registry/accessing-the-registry.adoc]: > ISG requires language like this when asking the user to enter a command. ("Commands, parameters, and options in instructions," p. 196, latest version)

Used 'run the following comman' adhering to ISG.

File: registry/accessing-the-registry.adoc (hunk 7 above)
Comment (snarayan-redhat) [reply registry/accessing-the-registry.adoc:43]: > This could just be a note instead of its own bullet.

SME suggested that this should be an explicit prerequisite for pushing images hence avoided putting this up as a note.

File: registry/accessing-the-registry.adoc (hunk 6 above)
Comment (snarayan-redhat) [reply registry/accessing-the-registry.adoc]: @maxwelldb I have made the suggested changes

**Review by maxwelldb — APPROVED** [review]

**Comment by maxwelldb** [comment]
TY! Refer to https://coreos.slack.com/archives/C85JYPHL3/p1626085853033600 for the latest on the merge freeze.

**Comment by mburke5678** [comment]
@snarayan-redhat LGTM
